applications/fishing-map/src/features/download/DownloadModal.tsx

```diff
@@ -0,0 +1,277 @@
+      dateRange: timerange as DateRange,
+      dataviews: downloadDataviews,
+      geometry: downloadAreaGeometry as DownloadGeometry,
+      areaName: downloadAreaName,
+      format,
+      temporalResolution,
+      spatialResolution,
+      groupBy,
+    }
+
+    try {
+      await dispatch(downloadActivityThunk(downloadParams))
+    } catch (e: any) {
+      console.warn(e)
+    }
+
+    uaEvent({
+      category: 'Download',
+      action: `Activity download`,
+      label: getEventLabel([
+        downloadAreaName,
+        ...downloadDataviews
+          .map(({ datasets, filters }) => [datasets.join(','), ...getActivityFilters(filters)])
+          .flat(),
+      ]),
+    })
+
+    timeoutRef.current = setTimeout(() => {
+      dispatch(resetDownloadStatus(undefined))
+    }, 5000)
+  }
+
+  return (
+    <Modal
+      title={t('download.title', 'Download - Activity')}
+      isOpen={isOpen}
+      onClose={onClose}
+      contentClassName={styles.modalContent}
+    >
+      <div className={styles.container}>
+        <div className={styles.info}>
+          <div>
+            <label>{t('download.area', 'Area')}</label>
+            <Tag>{downloadAreaName || EMPTY_FIELD_PLACEHOLDER}</Tag>
+          </div>
+          <div>
+            <label>{t('download.timeRange', 'Time Range')}</label>
+            <Tag>
+              <TimelineDatesRange />
+            </Tag>
+          </div>
+        </div>
+        <div>
+          <label>{t('download.format', 'Format')}</label>
+          <Choice
+            options={formatOptions}
+            size="small"
+            activeOption={format}
+            onOptionClick={(option) => setFormat(option.id as Format)}
+          />
+        </div>
+        {format === Format.Csv && (
+          <Fragment>
+            <div>
+              <label>{t('download.groupActivityBy', 'Group activity by')}</label>
+              <Choice
+                options={groupByOptions}
+                size="small"
+                activeOption={groupBy}
+                onOptionClick={(option) => setGroupBy(option.id as GroupBy)}
+              />
+            </div>
+            <div>
+              <label>{t('download.temporalResolution', 'Temporal Resolution')}</label>
+              <Choice
+                options={filteredTemporalResolutionOptions}
+                size="small"
+                activeOption={temporalResolution}
+                onOptionClick={(option) => setTemporalResolution(option.id as TemporalResolution)}
+              />
+            </div>
+          </Fragment>
+        )}
+        <div>
+          <label>{t('download.spatialResolution', 'Spatial Resolution')}</label>
+          <Choice
+            options={filteredSpatialResolutionOptions}
+            size="small"
+            activeOption={spatialResolution}
+            onOptionClick={(option) => setSpatialResolution(option.id as SpatialResolution)}
+          />
+        </div>
```

Comment:
I suggest having the spatial resolution choice first so that when toggling between CSV and GeoTIFF this stays in place.
```suggestion
        <div>
          <label>{t('download.spatialResolution', 'Spatial Resolution')}</label>
          <Choice
            options={filteredSpatialResolutionOptions}
            size="small"
            activeOption={spatialResolution}
            onOptionClick={(option) => setSpatialResolution(option.id as SpatialResolution)}
          />
        </div>
        {format === Format.Csv && (
          <Fragment>
            <div>
              <label>{t('download.groupActivityBy', 'Group activity by')}</label>
              <Choice
                options={groupByOptions}
                size="small"
                activeOption={groupBy}
                onOptionClick={(option) => setGroupBy(option.id as GroupBy)}
              />
            </div>
            <div>
              <label>{t('download.temporalResolution', 'Temporal Resolution')}</label>
              <Choice
                options={filteredTemporalResolutionOptions}
                size="small"
                activeOption={temporalResolution}
                onOptionClick={(option) => setTemporalResolution(option.id as TemporalResolution)}
              />
            </div>
          </Fragment>
        )}
```